### PR TITLE
feat: create mayastor deployment script for aks

### PR DESCRIPTION
**Pull request by abhisheksinghbaghel**
In this PR:
    1. Create deployment script for aks
    2. Use managed disk for etcd deployment
    3. Fix few minor issues with huge page enabler

### CODE REVIEW

File: scripts/deploy/aks-install-mayastor.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+readonly PKG_ROOT="$(git rev-parse --show-toplevel)"
+
+kubectl label nodes --selector agentpool=agentpool openebs.io/engine=mayastor --overwrite
```
Comment (landreasyan) [inline scripts/deploy/aks-install-mayastor.sh:20]: this command doesn't work for me. I have been using the following instead: 
kubectl label nodes -l agentpool=agentpool openebs.io/engine=mayastor --overwrite

Comment (landreasyan) [inline scripts/deploy/aks-install-mayastor.sh:20]: When we label all the agent nodes to be "storage nodes" with the current cluster setup, there is not enough CPU left to run the core agent. We either should not mark all the nodes as storage, i.e. openebs.io/engine=mayastor or we could increase the CPU for the nodes.

Comment (abhisheksinghbaghel) [reply scripts/deploy/aks-install-mayastor.sh:20]: -l, --selector='': Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l
key1=value1,key2=value2).

Comment (landreasyan) [reply scripts/deploy/aks-install-mayastor.sh:20]: by cluster setup I am referring to the [single-az.json](https://github.com/abhisheksinghbaghel/mayastor/blob/develop/scripts/deploy/cluster/single-az/single-az.json) script Akash has added to the repo.

Comment (abhisheksinghbaghel) [reply scripts/deploy/aks-install-mayastor.sh:20]: For now, let's just use a bigger SKU.

Comment (nearora-msft) [reply scripts/deploy/aks-install-mayastor.sh:20]: I faced the same issue that @landreasyan mentioned. The core-agents pod takes up a lot of CPU and on whichever node that is running and mayastor pod is not able to run on that node. We should probably run the control plane(core-agents) and data plane on different nodes.

Comment (abhisheksinghbaghel) [reply scripts/deploy/aks-install-mayastor.sh:20]: In an ideal set-up like AKS, we will run control-plane separately. However, when we start asking for separate nodes for each components, then we are moving away from the support in smaller clusters. 
For now lets use bigger SKU to unblock, but in a longer run we should strive to be able to run all this on a single node given minimum required resources.

**Pull request merged**